profiles/views.py

```diff
@@ -55,41 +54,18 @@ class SavedCompaniesCreate(CreateAPIView):
     serializer_class = SavedCompanySerializer
     pagination_class = ForumPagination
 
-    def post(self, request):
-        user = request.user
-        pk = request.data.get("company_pk")
-
-        # Check if the company is already in the user's saved list
-        if SavedCompany.objects.filter(user=user, company_id=pk).exists():
-            saved_company_destroyer = SavedCompaniesDestroy()
-            return saved_company_destroyer.destroy(request, pk)
-
-        serializer = SavedCompanySerializer(
-            data={"company": pk, "user": user.id}
-        )
-        if serializer.is_valid():
-            serializer.save()
-            return Response({"Company added": serializer.data})
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    def perform_create(self, serializer):
+        return serializer.save(user=self.request.user)
 
 
 class SavedCompaniesDestroy(DestroyAPIView):
     """
     Remove the company from the saved list.
     """
 
+    queryset = SavedCompany.objects.all()
     permission_classes = [IsAuthenticated]
 
```

Comment:
Please note that SavedCompaniesDestroy in your implementation performs a lookup by 'pk' (i.e., the subscription id), as it does by default in DestroyAPIView. However, on the frontend, we don't have this information, we need a lookup by the combination of the user id and the company id. If I were you, I would override the get_object() method for SavedCompaniesDestroy.